src/shared/src/impls.rs

```diff
@@ -1,9 +1,9 @@
-use crate::types::custom_token::{UserToken, UserTokenId};
+use crate::types::custom_token::{CustomToken, UserTokenId};
 
-impl From<UserToken> for UserTokenId {
-    fn from(token: UserToken) -> Self {
+impl From<CustomToken> for UserTokenId {
```

Comment:
I would recommend `impl From<&CustomToken> for ...` so that the token doesn't  need to be cloned every time you want to get the `TokenId`.  My bad, I probably omitted the `&` in my sketch.

Reply:
Good idea! Done.